alica_engine/include/engine/AlicaContext.h

```diff
@@ -201,14 +201,21 @@ class AlicaContext
      * Initialize alica framework and related modules.
      *
      * @param creatorCtx Creator functions for utility, behaviour, constraint and condition
-     * @param delayStarted does not start _engine. Set to true only for testing purpose
+     * @param delayStarted does not start _engine. To start engine it is possible to call startEngine()
+     *  If you need to add a worldmodel set this to true and startEngine() after.
      *
      * @return Return code '0' stands for success, any other for corresponding error
      *
      * @see AlicaCreators
      */
     int init(AlicaCreators&& creatorCtx, bool delayStart = false);
 
+    /**
+     * Start engine if not previously started (Do you need to add a worldmodel?)
+     *
+     */
+    void startEngine();
```

Comment:
can't the blackboard be stored in the AlicaContext instead & the world model set by some init plan in the library? Then this function can be removed